src/modules/DASDController.rb

```diff
@@ -137,7 +147,17 @@ def Write
             to_reactivate << device
           end
           if format
-            to_format << (device["dev_name"] || GetDeviceName(channel))
+            if can_be_formatted?(device)
+              to_format << (device["dev_name"] || GetDeviceName(channel))
```

Comment:
That `(device["dev_name"] || GetDeviceName(channel))` is everywhere. In this case even in both branches of an `if/else`.